src/_experimental_/GraphQL/QlHooks.re

```diff
@@ -405,6 +427,34 @@ let useTotalCollectedToken: Animal.t => option((Eth.t, BN.bn, BN.bn)) =
     | _ => None
     };
   };
+type patronLoyaltyTokenDetails = {
+  currentLoyaltyTokens: Eth.t,
+  lastCollected: BN.bn,
+  numberOfAnimalsOwned: BN.bn,
+};
+let usePatronLoyaltyTokenDetails:
+  Web3.ethAddress => option(patronLoyaltyTokenDetails) =
+  address => {
+    let (response, _) = useQueryPatronNew(address);
+
+    switch (response) {
+    | Data(responseData) =>
+      responseData##patronNew
+      ->Belt.Option.flatMap(patron =>
+          Some({
+            currentLoyaltyTokens: BN.new_("123"),
+            lastCollected: patron##lastUpdated,
+            numberOfAnimalsOwned:
+              BN.new_(patron##tokens->Obj.magic->Array.length->string_of_int),
```

Comment:
I just checked, the `tokens` parameter on `patronsNew` is incorrect. So that is a blocker for this PR at the moment.